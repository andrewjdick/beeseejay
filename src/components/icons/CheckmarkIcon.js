import React from "react";
import { Icon } from "./Icon";

export const CheckmarkIcon = props => (
  <Icon {...props}>
    <path d="M2 20 L12 28 30 4" />
  </Icon>
);
