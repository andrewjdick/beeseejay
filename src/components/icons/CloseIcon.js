import React from "react";
import { Icon } from "./Icon";

export const CloseIcon = props => (
  <Icon {...props}>
    <path d="M2 30 L30 2 M30 30 L2 2" />
  </Icon>
);
