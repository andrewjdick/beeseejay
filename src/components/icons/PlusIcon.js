import React from "react";
import { Icon } from "./Icon";

export const PlusIcon = props => (
  <Icon {...props}>
    <path d="M16 2 L16 30 M2 16 L30 16" />
  </Icon>
);
