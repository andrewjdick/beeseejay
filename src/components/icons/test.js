import React from "react";
import renderer from "react-test-renderer";
import { CheckmarkIcon } from "./CheckmarkIcon";
import { CloseIcon } from "./CloseIcon";
import { PlusIcon } from "./PlusIcon";
import { TrashIcon } from "./TrashIcon";

it("renders all icons correctly", () => {
  const checkmarkIcon = renderer.create(<CheckmarkIcon />).toJSON();
  const closeIcon = renderer.create(<CloseIcon />).toJSON();
  const plusIcon = renderer.create(<PlusIcon />).toJSON();
  const trashIcon = renderer.create(<TrashIcon />).toJSON();

  expect(checkmarkIcon).toMatchSnapshot();
  expect(closeIcon).toMatchSnapshot();
  expect(plusIcon).toMatchSnapshot();
  expect(trashIcon).toMatchSnapshot();
});
