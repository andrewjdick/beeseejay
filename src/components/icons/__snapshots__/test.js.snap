// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders all icons correctly 1`] = `
<svg
  className="sc-bdVaJa iKBuLo"
  fill="none"
  height={32}
  id="i-close"
  stroke="currentcolor"
  strokeLinecap="round"
  strokeLinejoin="round"
  strokeWidth="2"
  viewBox="0 0 32 32"
  width={32}
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M2 20 L12 28 30 4"
  />
</svg>
`;

exports[`renders all icons correctly 2`] = `
<svg
  className="sc-bdVaJa iKBuLo"
  fill="none"
  height={32}
  id="i-close"
  stroke="currentcolor"
  strokeLinecap="round"
  strokeLinejoin="round"
  strokeWidth="2"
  viewBox="0 0 32 32"
  width={32}
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M2 30 L30 2 M30 30 L2 2"
  />
</svg>
`;

exports[`renders all icons correctly 3`] = `
<svg
  className="sc-bdVaJa iKBuLo"
  fill="none"
  height={32}
  id="i-close"
  stroke="currentcolor"
  strokeLinecap="round"
  strokeLinejoin="round"
  strokeWidth="2"
  viewBox="0 0 32 32"
  width={32}
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M16 2 L16 30 M2 16 L30 16"
  />
</svg>
`;

exports[`renders all icons correctly 4`] = `
<svg
  className="sc-bdVaJa iKBuLo"
  fill="none"
  height={32}
  id="i-close"
  stroke="currentcolor"
  strokeLinecap="round"
  strokeLinejoin="round"
  strokeWidth="2"
  viewBox="0 0 32 32"
  width={32}
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M28 6 L6 6 8 30 24 30 26 6 4 6 M16 12 L16 24 M21 12 L20 24 M11 12 L12 24 M12 6 L13 2 19 2 20 6"
  />
</svg>
`;
